packages/togetherai/src/togetherai-provider.ts

```diff
@@ -0,0 +1,140 @@
+import { LanguageModelV1, EmbeddingModelV1 } from '@ai-sdk/provider';
+import { OpenAICompatibleProviderSettings } from '@ai-sdk/openai-compatible';
+import {
+  OpenAICompatibleChatLanguageModel,
+  OpenAICompatibleCompletionLanguageModel,
+  OpenAICompatibleEmbeddingModel,
+} from '@ai-sdk/openai-compatible';
+import {
+  FetchFunction,
+  loadApiKey,
+  withoutTrailingSlash,
+} from '@ai-sdk/provider-utils';
+import {
+  TogetherAIChatModelId,
+  TogetherAIChatSettings,
+} from './togetherai-chat-settings';
+import {
+  TogetherAIEmbeddingModelId,
+  TogetherAIEmbeddingSettings,
+} from './togetherai-embedding-settings';
+import {
+  TogetherAICompletionModelId,
+  TogetherAICompletionSettings,
+} from './togetherai-completion-settings';
+
+export interface TogetherAIProviderSettings
+  extends OpenAICompatibleProviderSettings {
+  /**
+   TogetherAI API key.
+   */
+  apiKey?: string;
```

Comment:
Add baseURL & headers (important for proxies and every provider has them for that reason)

Reply:
We had these via extension of `OpenAICompatibleProviderSettings` but I get now that it's better to have a copy as we avoid leaking settings we don't want to expose in TogetherAI's settings. Changed to copy those (and `fetch`).